packages/java/endpoint/src/main/java/dev/hilla/crud/filter/Remapper.java

```diff
@@ -0,0 +1,115 @@
+package dev.hilla.crud.filter;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+
+/**
+ * Remaps property names in filters and pageable objects.
+ */
+public class Remapper {
+
+    private final Map<String, String> mappings = new HashMap<>();
+    private Function<PropertyStringFilter, PropertyStringFilter> filterTransformation;
+
+    /**
+     * Declares a mapping from one property name to another. If a filter or
+     * pageable is remapped, all occurrences of the property name will be
+     * replaced with the new name.
+     *
+     * @param from
+     *            The original property name.
+     * @param to
+     *            The new property name.
+     * @return This instance.
+     */
+    public Remapper withMapping(String from, String to) {
+        mappings.put(from, to);
+        return this;
+    }
+
+    /**
+     * Declares a function that will be applied to all
+     * {@link PropertyStringFilter} instances. This can be used to modify the
+     * filter value in a more complex way than a simple mapping.
```

Comment:
```suggestion
     * Declares a function that will be applied to all
     * {@link PropertyStringFilter} instances. This can be used to modify the
     * filter value in a more complex way than a simple mapping.
     * <p>
     * Note: The passed in {@code filterTransformation} function is applied
     * after all other mappings are applied.
```

Reply:
Another comment that comes to mind is that the `filterTransformation` function being applied on top of the `dto -> entity` mappings may not be very intuitive. Providing support a custom way of transformation for corner cases may be a good idea, but, doing it on top of simple mappings makes it complicated IMO. Developer need to keep it in mind that though this is a filter transformation function, they may need to rely on property names from entity depend on what mappings applied beforehand. I'd suggest simplify it so that it only support simple property name/path mappings from DTO to ENTITY.

Reply:
Maybe we can provide factory methods to instantiate a `Remapper` instance either with simple mappings or based on a passed in transformation function.